sql-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/RapidsShuffleIterator.scala

```diff
@@ -152,6 +132,8 @@ class RapidsShuffleIterator(
 
   private var started: Boolean = false
 
+  case class BlockIdMapIndex(id: ShuffleBlockBatchId, mapIndex: Int)
```

Comment:
Curious, how is `mapIndex` different than `id.mapId`?  If it's different a comment explaining would help.

Reply:
`mapId` used to be `mapIndex`, but it is now `taskAttemptId` as long as you are using the new shuffle protocol (default ON). `mapIndex` is the index into the `statuses` array in `MapOutputTracker`.

`reduceId` is the index into the `sizes` array, within a `MapStatus`.

This function summarizes it pretty well I think: https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/MapOutputTracker.scala#L949.

I'll add a comment as suggested.